### ISSUE TITLE
Sim sled-agent: Stopped instances lose their disks

### ISSUE THREAD

**Issue by askfongjojo** [oxidecomputer/omicron#2803]
The sim sled-agent now marks disks as detached when an instance is in the `stopped` state because of the fix for  https://github.com/oxidecomputer/omicron/issues/2675 ( https://github.com/oxidecomputer/omicron/commit/207768ea2bdc644da2c5c7b5f7ababab4a6f4e95). An instance in `stopped` state should probably still own the disks (although they are not mounted/attached physically). If not, when the instance is started again, it will have no disk.

@smklein, @jmpesp  - The change is only confined to the simulated sled-agent. Does it reflect how a real sled would work in the stopped and destroyed scenarios?

**Comment by jmpesp** [comment]
You're right - for the real sled agent, stopping an instance does not detach the disks. This is a bug introduced by me for the simulated sled agent.
